src/EditorFeatures/Core.Wpf/Preview/DifferenceViewerPreview.cs

```diff
@@ -0,0 +1,235 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Interop;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.VisualStudio;
+using Microsoft.VisualStudio.OLE.Interop;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Text.Differencing;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.Preview
+{
+    internal sealed partial class DifferenceViewerPreview : IDifferenceViewerPreview<IWpfDifferenceViewer>
```

Comment:
Almost all the code in this file is a port from https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient/pullrequest/496210, while also retaining the prior code in https://github.com/dotnet/roslyn/blob/main/src/EditorFeatures/Core/Preview/DifferenceViewerPreview.cs